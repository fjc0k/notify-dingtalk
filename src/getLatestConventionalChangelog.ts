import fs from 'fs'
import { join, resolve } from 'path'

export function getLatestConventionalChangelog(
  conventionalChangelogFile = join(process.cwd(), './CHANGELOG.md'),
) {
  const content = fs.readFileSync(
    resolve(process.cwd(), conventionalChangelogFile),
    'utf8',
  )
  const latestChangelog =
    content.match(/(?:^|\n)(#+\s+\[?\d.+?)(?:\n#+\s+\[?\d|$)/s)?.[1] ?? ''
  return latestChangelog.trim()
}
