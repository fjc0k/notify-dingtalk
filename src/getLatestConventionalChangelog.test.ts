import { getLatestConventionalChangelog } from './getLatestConventionalChangelog'
import { join } from 'path'

const resolve = (file: string) => join(__dirname, file)

test('表现正常', () => {
  expect(
    getLatestConventionalChangelog(resolve('./__fixtures__/CHANGELOG1.md')),
  ).toMatchSnapshot('CHANGELOG1')
  expect(
    getLatestConventionalChangelog(resolve('./__fixtures__/CHANGELOG2.md')),
  ).toMatchSnapshot('CHANGELOG2')
  expect(
    getLatestConventionalChangelog(resolve('./__fixtures__/CHANGELOG3.md')),
  ).toMatchSnapshot('CHANGELOG3')
  expect(
    getLatestConventionalChangelog(resolve('./__fixtures__/CHANGELOG4.md')),
  ).toMatchSnapshot('CHANGELOG4')
  expect(
    getLatestConventionalChangelog(resolve('./__fixtures__/CHANGELOG5.md')),
  ).toMatchSnapshot('CHANGELOG5')
  expect(
    getLatestConventionalChangelog(resolve('./__fixtures__/CHANGELOG6.md')),
  ).toMatchSnapshot('CHANGELOG6')
  expect(
    getLatestConventionalChangelog(resolve('./__fixtures__/CHANGELOG7.md')),
  ).toMatchSnapshot('CHANGELOG7')
})
