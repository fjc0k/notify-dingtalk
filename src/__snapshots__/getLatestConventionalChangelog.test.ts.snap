// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`表现正常: CHANGELOG1 1`] = `
"## 1.0.0 (2020-12-11)

### Features

- 初始化 ([e68dba8](https://github.com/fjc0k/notify-dingtalk/commit/e68dba8e45e05126e6a0d648591a8a1a8a1e8c18))"
`;

exports[`表现正常: CHANGELOG2 1`] = `
"## [3.11.1](https://github.com/fjc0k/yapi-to-typescript/compare/v3.11.0...v3.11.1) (2020-12-11)

### Bug Fixes

- 修复 getFormData 错误 ([44df2ad](https://github.com/fjc0k/yapi-to-typescript/commit/44df2ad))

<a name=\\"3.11.0\\"></a>"
`;

exports[`表现正常: CHANGELOG3 1`] = `""`;

exports[`表现正常: CHANGELOG4 1`] = `
"## [1.10.0](http://foo.bar/pull-requests?create&targetBranch=refs%2Ftags%2Fv1.9.0&sourceBranch=refs%2Ftags%2Fv1.10.0) (2020-12-11)

### 特性

- 重构地址选择器、类目选择器等级联选择器以支持回显，所有级联选择器的绑定值现在都是选中最后一级的值 ([2726e09](http://foo.bar/commits/2726e098a113c7215329c8bdba5890caf921e737))

### 修复

- 解决控制台总是报属性缺失的问题 ([c29d1a6](http://foo.bar/commits/c29d1a65db6f993084578c3ab2195d13e5f65609))"
`;

exports[`表现正常: CHANGELOG5 1`] = `
"## [1.10.0-beta5](http://foo.bar/pull-requests?create&targetBranch=refs%2Ftags%2Fv1.9.0&sourceBranch=refs%2Ftags%2Fv1.10.0) (2020-12-11)

### 特性

- 重构地址选择器、类目选择器等级联选择器以支持回显，所有级联选择器的绑定值现在都是选中最后一级的值 ([2726e09](http://foo.bar/commits/2726e098a113c7215329c8bdba5890caf921e737))

### 修复

- 解决控制台总是报属性缺失的问题 ([c29d1a6](http://foo.bar/commits/c29d1a65db6f993084578c3ab2195d13e5f65609))"
`;

exports[`表现正常: CHANGELOG6 1`] = `
"### 0.0.1 (2020-11-16)

### Features

- 完成构建、发布流程 ([0151deb](http://http://foo.bar/commit/0151deb9d416ccc3902eabc3e0acdb8ed5fa7a24))
- **文档:** API ([0d0f086](http://http://foo.bar/commit/0d0f0865b7a9fccc804325c1843528894bfdd832))
- **组件:** 新增 AddressPicker 地址选择器 ([f073132](http://http://foo.bar/commit/f07313288c49a6bcf4898202789d7c57b6f51a0e))
- **组件:** 新增 MaterialPicker 物料选择器 ([089c11c](http://http://foo.bar/commit/089c11c8b5ce77aa569d9bc41eaa8f877f604583))
- 新增 renderComponent ([4346697](http://http://foo.bar/commit/43466971dfa898e6fce6708296aa6551c4fb0dd1))
- **组件:** 新增 StaffPicker 人员选择器 ([831a343](http://http://foo.bar/commit/831a343e08b8c9d6c829a9d45042b6dcdfdcbeba))
- 初始化项目 ([e30821e](http://http://foo.bar/commit/e30821e3ce3236b35948ff4e83861925ad423707))"
`;

exports[`表现正常: CHANGELOG7 1`] = `
"## [1.5.8](https://github.com/Darkhorse-Fraternity/combo/compare/v1.5.7...v1.5.8) (2020-12-25)

### Bug Fixes

- 添加 changelog ([960ac69](https://github.com/Darkhorse-Fraternity/combo/commit/960ac69dd00ccc40c690c38b22b6e2974866351e))"
`;
